addons/tmi/services/oauth.gd

```diff
@@ -250,8 +253,8 @@ func _code_to_token(code: String):
 		"image": payload.data.get("picture", "")
 	}
 	
-	newCredentials.channel = credentials.channel
-	newCredentials.broadcaster_user_id = await _lookup_channel(body.access_token)
+	newCredentials.channel = credentials.channel if !credentials.channel.is_empty() else payload.data.get("preferred_username", "")
+	newCredentials.broadcaster_user_id = await _lookup_channel(body.access_token, newCredentials.user_login)
```

Comment:
this be using the `newCredentials.channel` value so it reflects the channel being watched instead of the access token's user